sdk/cosmos/azure-cosmos-kafka-connect/src/main/java/com/azure/cosmos/kafka/connect/CosmosSourceConnector.java

```diff
@@ -167,13 +171,37 @@ private IMetadataReader getMetadataReader() {
                             throw new IllegalStateException("Cosmos Metadata container need to be partitioned by /id");
                         }
                     })
+                    .onErrorResume(throwable -> {
+                        if (KafkaCosmosExceptionsHelper.isNotFoundException(throwable)
+                            && shouldCreateMetadataContainerIfNotExists()) {
+                            return createMetadataContainer();
+                        }
+
+                        return Mono.error(new ConnectException(throwable));
+                    })
                     .block();
                 return new MetadataCosmosStorageManager(metadataContainer);
             default:
                 throw new IllegalArgumentException("Metadata storage type " + this.config.getMetadataConfig().getStorageType() + " is not supported");
         }
     }
 
+    private boolean shouldCreateMetadataContainerIfNotExists() {
+        // If customer does not create the metadata container ahead of time,
+        // then SDK will create one with default autoScale config only if using masterKey auth.
```

Comment:
I recall Fabric team mentioning to go ahead with default 400 RUs, no? I know we have seen problems with throttling with 400 RUs, but curious are they blockers?

Reply:
they proposed using 400RUs, but we decided to use autoscale 4000RUs